### PR TITLE
pkg/aws: check cluster name and worker pool name length

### LINKED ISSUE #179: Validate max worker pool name length and cluster name length in AWs
As since #151, the maximum length is now limited, so using long cluster names or worker pool names results in following error:
```
module.aws-lokomotive-mateusz.module.workers.data.aws_ami.flatcar[0]: Refreshing state...

Error: "name" cannot be longer than 32 characters

  on ../lokomotive-kubernetes/aws/flatcar-linux/kubernetes/workers/ingress.tf line 26, in resource "aws_lb_target_group" "workers-https":
  26: resource "aws_lb_target_group" "workers-https" {


FATA[0010] error applying cluster: failed checking execution status: exit status 1  args="[]" command="lokoctl cluster apply"
```

Also, this affects `examples` configuration we provide.

### PR DESCRIPTION
**Pull request by iaguis**
There are tight restrictions on resource name length on AWS. Check
cluster name and worker pool name against those.

Fixes #179

### CODE REVIEW

File: pkg/platform/aws/aws.go [hunk 1]
```diff
@@ -226,6 +226,35 @@ func (c *config) checkValidConfig() hcl.Diagnostics {
...
+
+	return diagnostics
+}
+
+// checkNameSizes checks the size of names since AWS has a limit of 32
+// characters on resources.
+func (c *config) checkNameSizes() hcl.Diagnostics {
+	var diagnostics hcl.Diagnostics
+
+	maxNameLen := 18 // 32 - len("-workers-https"). This is the longest resource suffix.
```
Comment (invidian) [inline pkg/platform/aws/aws.go]: Can we have this calculation in place rather than in the comment?

File: pkg/platform/aws/aws.go [hunk 2]
```diff
@@ -226,6 +226,35 @@ func (c *config) checkValidConfig() hcl.Diagnostics {
...
+
+	if len(c.ClusterName) > maxNameLen {
+		diagnostics = append(diagnostics, &hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Cluster name too long",
+			Detail:   "Maximum lenth is 18",
```
Comment (invidian) [inline pkg/platform/aws/aws.go]: should use `fmt.Sprintf` to include the max length variable.

File: pkg/platform/aws/aws.go [hunk 3]
```diff
@@ -226,6 +226,35 @@ func (c *config) checkValidConfig() hcl.Diagnostics {
...
+		})
+	}
+
+	for _, wp := range c.WorkerPools {
+		if len(wp.Name) > maxNameLen {
+			diagnostics = append(diagnostics, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Worker pool name too long",
+				Detail:   "Maximum lenth is 18",
```
Comment (invidian) [inline pkg/platform/aws/aws.go]: should use `fmt.Sprintf` to include the max length variable.

**Comment by iaguis** [comment]
> Does worker pool name combines with cluster name in any place? If not, is it possible to have 2 worker pools with the same name on 2 different clusters? If so, should we also add a check for it's maximum length?

I don't think it does, I've tested with both cluster name and worker pool name being 18 characters and it worked fine.

I think having two worker pools with the same name results in errors like the one solved in https://github.com/kinvolk/lokomotive/pull/171

File: pkg/platform/aws/aws.go (hunk 1 above)
Comment (iaguis) [reply pkg/platform/aws/aws.go]: Sure, I'll have to tame the linter though.

File: pkg/platform/aws/aws.go (hunk 2 above)
Comment (iaguis) [reply pkg/platform/aws/aws.go]: Right, thanks for the catch.

File: pkg/platform/aws/aws.go (hunk 3 above)
Comment (iaguis) [reply pkg/platform/aws/aws.go]: Right, thanks for the catch.

**Comment by invidian** [comment]
>I think having two worker pools with the same name results in errors like the one solved in #171

I wonder if we should document/report that :thinking:

**Comment by iaguis** [comment]
> I wonder if we should document/report that thinking

I created https://github.com/kinvolk/lokomotive/pull/186

**Comment by ipochi** [comment]
> > I think having two worker pools with the same name results in errors like the one solved in #171
> 
> I wonder if we should document/report that

In the same cluster , if there are two worker pools with same name, then there an appropriate error message provided.

**Pull request merged**